### PR TITLE
Restore the feature flags suite under mixed version testing

### PR DESCRIPTION
**Pull request by pjk25**
Previously when CI switched to bazel, these tests were skipped because they relied on functions only exported for test, and bazel mixed version testing uses non-dev builds of rabbit for the secondary umbrella. Recent changes to the suite mean that this is no longer the case.

### CODE REVIEW

**Comment by pjk25** [comment]
@mergifyio rebase

**Comment by pjk25** [comment]
Will need a rebase again after #5302 is merged

**Comment by pjk25** [comment]
@mergifyio rebase

**Comment by pjk25** [comment]
#5302 has now been merged, but I think it would be sensible to get master back to green before this can be merged (provided it does not need further corrections itself).

**Review by dumbbell — APPROVED** [review]

**Comment by pjk25** [comment]
Now that this has been rebased against the other changes, it looks like there might be an issue loading https://github.com/rabbitmq/inet_tcp_proxy on the mixed version nodes. I'm investigating.

**Comment by pjk25** [comment]
inet_tcp_proxy was a red herring. One of the failures is due to `rabbit_feature_flags:inject_test_feature_flags/1` being only exported for test on all current releases. I will think a little bit about how to best address/workaround the limitation.

**Comment by pjk25** [comment]
This will require #5336

**Comment by dumbbell** [comment]
I just force-pushed the branch with two more fixes and rebased on the latest `master` (which includes #5336). It passes successfully locally for me, both in regular and mixed-version testing.

File: deps/rabbit/test/feature_flags_SUITE.erl
Comment (pjk25) [inline deps/rabbit/test/feature_flags_SUITE.erl:126]: I'm going to validate that this method of checking will also work with bazel, as there are not secondary sources present

File: deps/rabbit/test/feature_flags_SUITE.erl
Comment (dumbbell) [inline deps/rabbit/test/feature_flags_SUITE.erl:129]: The order of checks was reversed in this patch.

The use of a secondary umbrella should be checked first. If there is none configured, the `feature_flags_v2` test group should be executed regardless of the support. Because the test group was committed with the code, it's an error if the test group runs without the corresponding code in a non-mixed-version scenario.

File: deps/rabbit/test/feature_flags_SUITE.erl
Comment (dumbbell) [inline deps/rabbit/test/feature_flags_SUITE.erl:142]: This will work with Make as well.

Beside the paths to `File1` and `File2`, the rest of the code should be common to Make and Bazel. This should simplify it.

File: deps/rabbit/test/feature_flags_SUITE.erl
Comment (dumbbell) [inline deps/rabbit/test/feature_flags_SUITE.erl:175]: Using a secondary umbrella works with swapped versions too (primary == old, secondary == new), so the wording is incorrect.

File: deps/rabbit/test/feature_flags_SUITE.erl
Comment (dumbbell) [inline deps/rabbit/test/feature_flags_SUITE.erl:167]: Once the checks are re-ordered, this clause should be removed as it will never happen.

**Review by dumbbell — CHANGES_REQUESTED** [review]

**Review by dumbbell — CHANGES_REQUESTED** [review]
Could you please split this commit into two? Both chunks share nothing.

**Review by dumbbell — APPROVED** [review]

**Pull request merged**
